c/prima.c

```diff
@@ -161,43 +125,12 @@ int prima_free_result(prima_result_t *result)
 /* these functions just call the fortran compatibility layer and return the status code */
 int prima_minimize(const prima_algorithm_t algorithm, prima_problem_t *problem, prima_options_t *options, prima_result_t *result)
 {
-  int alloc_bounds = (algorithm == PRIMA_COBYLA) || (algorithm == PRIMA_BOBYQA) || (algorithm == PRIMA_LINCOA);
   int use_constr = (algorithm == PRIMA_COBYLA);
 
-  int info = prima_check_problem(problem, options, alloc_bounds, use_constr);
+  int info = prima_check_problem(problem, options, use_constr);
   if (info == 0)
     info = prima_init_result(result, problem);
 
-  if ((info == 0) && use_constr)
-  {
-    // reuse or (re)allocate nlconstr
-    if (result->_m_nlcon != problem->m_nlcon)
```

Comment:
This particular if statement relied on the fact that `result->_m_nlcon` would be 0 after the call to `prima_init_result`. I moved it into `prima_init_result` and relied on the value of `problem->m_nlcon` to allocate it.